.github/workflows/build-docker-images.yml

```diff
@@ -0,0 +1,124 @@
+
+env:
+  AWS_REGION: ${{ inputs.awsRegion }}
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-images:
+    strategy:
+      matrix:
+        image: [
+          {
+            name: "bfd-mgmt-eft-sftp-outbound-transfer-lambda", 
+            dockerfile: "ops/jenkins/bfd-build-eft-sftp-outbound-transfer-lambda/Dockerfile", 
+            contextDir: "ops/terraform/services/eft/lambda_src/sftp_outbound_transfer"
+          }
+        ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate Inputs
+        run: |
+          echo "Validating inputs to ensure they conform to expected formats..."
+          echo "${{ inputs.versionTag }}" | grep -P '^\d+\.\d+\.\d+$|^\d+\.\d+\.\d+-[a-zA-Z0-9-]+$'
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ inputs.branch }}
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.GHA_AWS_IAM_ROLE_ARN }}
+          role-session-name: build-images
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Login to ECR
+        uses: aws-actions/amazon-ecr-login@v2
+          
+      - name: Login to Amazon ECR Public
+        id: login-ecr-public
+        uses: aws-actions/amazon-ecr-login@v2
+        with:
+          registry-type: public
+
+      - name: Get ECR Registry Namespace
+        run: |
+          ECR_REPOSITORY_NAMESPACE="$(aws ecr describe-registry --region "$AWS_REGION" | jq -r '.registryId').dkr.ecr.${AWS_REGION}.amazonaws.com"
+          echo "::add-mask::$ECR_REPOSITORY_NAMESPACE"
+          echo ECR_REPOSITORY_NAMESPACE=$ECR_REPOSITORY_NAMESPACE >> $GITHUB_ENV
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build and Push with Latest Tag
+        # Naively, only build and push with latest if the versionTag is a release version
+        if: ${{ !contains(inputs.versionTag, '-') }}
+        uses: docker/build-push-action@v5
+        with:
+          file: ${{ matrix.image.dockerfile }}
+          context: ${{ matrix.image.contextDir }}
+          push: true
+          tags: |
+            ${{ env.ECR_REPOSITORY_NAMESPACE }}/${{ matrix.image.name }}:latest
+            ${{ env.ECR_REPOSITORY_NAMESPACE }}/${{ matrix.image.name }}:${{ inputs.versionTag }}
+        env:
+          DOCKER_BUILDKIT: 1
+
+      - name: Build and Push without Latest Tag
+        # Naively, build and push without latest tag if the versionTag is not a release version
+        if: ${{ contains(inputs.versionTag, '-') }}
+        uses: docker/build-push-action@v5
+        with:
+          file: ${{ matrix.image.dockerfile }}
+          context: ${{ matrix.image.contextDir }}
+          push: true
+          tags: |
+            ${{ env.ECR_REPOSITORY_NAMESPACE }}/${{ matrix.image.name }}:${{ inputs.versionTag }}
+        env:
+          DOCKER_BUILDKIT: 1
```

Comment:
I am unsure how to reduce this duplication. If anyone understands GitHub Actions better than I, please let me know how to make this better.

Reply:
This has been resolved by using an intermediate step to generate the list of tags conditionally, then passing that list to the `docker/build-push-action` via a step output.